lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2778,18 +2778,7 @@ bool SubfieldOp::isFieldFlipped() {
 FIRRTLType SubindexOp::inferReturnType(ValueRange operands,
                                        ArrayRef<NamedAttribute> attrs,
                                        std::optional<Location> loc) {
-  Type inType = operands[0].getType();
-  auto fieldIdx =
-      getAttr<IntegerAttr>(attrs, "index").getValue().getZExtValue();
-
-  if (auto vectorType = inType.dyn_cast<FVectorType>()) {
-    if (fieldIdx < vectorType.getNumElements())
-      return vectorType.getElementType();
-    return emitInferRetTypeError(loc, "out of range index '", fieldIdx,
-                                 "' in vector type ", inType);
-  }
-
-  return emitInferRetTypeError(loc, "subindex requires vector operand");
+  return operands[0].getType().cast<FVectorType>().getElementType();
```

Comment:
I'm not sure where to write this, but it seems like we're forming a sort of implicit rule for where checks (should) happen:

1) Constraints where possible, they're checked first (possibly dependent on trait order, or maybe not, didn't chase this down?)
2) inferReturnTypes for most things, since `verify` is hard to reach if this fails (even if type inference sometimes cause an LLVM fatal error :crying_cat_face: ).  If covered by (1) may be safe to omit other than cases where parse uses `inferReturnTypes` as verifier.  EDIT: maybe not, see below.
3) verify as well for completeness in case we're missing something, unsure how to reach this for operations with `inferReturnTypes` methods but seems of all places this is where non-constraint checks should go?

Reply:
Unfortunately, we need these checks for reasons mentioned in (2) above?

For example, this input previously produced neat error "subindex requires vector operand" but now crashes:

```firrtl
circuit invalid_name :
  module invalid_name :
   input out_0 : SInt<8>
   out_0[4] <- out_0[5]
```